src/bindable-property.js

```diff
@@ -35,7 +35,13 @@ export class BindableProperty {
   * Creates an instance of BindableProperty.
   * @param nameOrConfig The name of the property or a cofiguration object.
   */
-  constructor(nameOrConfig: string | Object) {
+  constructor(nameOrConfig: string | {
+    defaultBindingMode?: number,
+    reflect?: boolean | {(el: Element, name: string, newVal, oldVal): any},
```

Comment:
I'm not convinced by the API design of the function part.
It feels like it takes too much... the element and old value and a name (of what)?

This is meant to just reflect a value in an attribute. The callback here really feels like `changeHandler`. Accepting a `bool | string` with the string allowing to reflect to a different attribute name seems simpler.

Reply:
If we restricted it to only choose the name, whjat would we do if it is not always 1:1 prop->attr ?

Reply:
Do you mean like one binding updates 2 attributes? For example `condition` boolean property is reflected into the presence of either an `true` or a `false` attribute on target element?

Personally I would say this is narrow use cases that can be filled today by using `changeHandler`, which has practically the same signature.

Reply:
It's the case where value of property is an object, we dont want `[object Object]` in the attribute.

Another motivation, is that pretty much everything in Aurelia is intercept-able, So I would choose go with the rest on this.